### DISCUSSION TITLE
SplitChunks config but for Worker script only?

### DISCUSSION THREAD

**Discussion by thomastay** [webpack/webpack#17657] [Q&A]
Hi, is it possible to apply a certain set of optimization.splitChunks rules to only a Web Worker script?

e.g. in our main app, we have the following config
```js
{
  optimization: {
    splitChunks: {
      chunks: "all",
      cacheGroups: { /* some cache groups here that we want to split out, e.g. Lodash */ }
    }
  }
}
```
This splits out the main app into many initial chunks which is nice. The cacheGroups works well for us here too.

But in the worker, we don't want any initial chunks at all, since the worker has to run `importScripts`, which is a synchronous call. Even if we have to duplicate code used in both main app and worker script, we are ok with that.

I know you can do this:
```js
{
  optimization: {
    splitChunks: {
      chunks: chunk => chunk.name != "MY_WORKER_CHUNK_NAME"
      cacheGroups: { /* some cache groups here that we want to split out, e.g. Lodash */ }
    }
  }
}
```
But we were wondering if there's any other way. Also, the cacheGroups (e.g. Lodash) still gets split out of the Worker script into a separate bundle, which we don't want. 
Again, we're fine with duplicating code.
I guess what I really want is to treat the Worker like an entrypoint, which it currently isn't.

Any ideas how I can make that work?

**Comment by alexander-akait** [comment]
> I guess what I really want is to treat the Worker like an entrypoint, which it currently isn't.

Using https://webpack.js.org/concepts/entry-points/#entrydescription-object or using multi compiler mode (i.e. `module.exports = [webpackConfigForApplication, webpackConfigForWorker]`)
